### PR TITLE
Update protobuf to 21.9

### LINKED ISSUE #22686: Vulnerabilities in protobuf-java 3.19.4
[CVE-2022-3171](https://nvd.nist.gov/vuln/detail/CVE-2022-3171)

It was already fixed for 5.2.z and reported here: https://github.com/hazelcast/hazelcast/issues/22541, but we need backport of https://github.com/hazelcast/hazelcast/pull/22544 in: `5.1.z` `5.0.z` `master`

### PR DESCRIPTION
**Pull request by frant-hartm**
Protobuf has changed the versioning scheme. Each language has its own major version - Java packages stayed on `3.x`, but python has moved to `4.x`.

Use two properties - `java.protobuf.version` and `python.protobuf.version` derived from `protobuf.version` property.

Recreated the `jet_to_python_pb2.py` file by running the protobuf plugin.

Fixes #22686 on master

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: extensions/python/pom.xml [hunk 1]
```diff
@@ -101,13 +101,13 @@
                             <execution>
                                 <id>python</id>
                                 <configuration>
-                                    <protocArtifact>com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}</protocArtifact>
+                                    <protocArtifact>com.google.protobuf:protoc:${java.protobuf.version}:exe:${os.detected.classifier}</protocArtifact>
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply extensions/python/pom.xml:104]: It points to this maven artifact
https://mvnrepository.com/artifact/com.google.protobuf/protoc/3.21.9
so it matches the java version.

[the remark this answers is not in the document]
Comment (frant-hartm) [reply extensions/python/pom.xml:104]: The python version is for packages from pip
https://pypi.org/project/protobuf/#history

**Comment by frant-hartm** [comment]
run-lab-run

File: pom.xml [hunk 2]
```diff
@@ -119,7 +119,15 @@
         <picocli.version>4.6.3</picocli.version>
         <postgresql.version>42.4.2</postgresql.version>
         <prometheus.version>0.17.1</prometheus.version>
-        <protobuf.version>3.19.4</protobuf.version>
+
+        <!--
+        Protobuf libraries for each language have own major version, but share minor and patch
+        See https://developers.google.com/protocol-buffers/docs/news/2022-05-06
```
Comment (ldziedziul) [inline pom.xml:125]: 😮

**Review by ldziedziul — APPROVED** [review]
Looks good, I think ;)

**Review by orcunc — APPROVED** [review]

**Pull request merged**